lib/ibm_cloud_sdk_core/icp4d_token_manager.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+require("http")
+require("json")
+require("rbconfig")
+require_relative("./version.rb")
+require_relative("./jwt_token_manager")
+
+module IBMCloudSdkCore
+  # Class to manage ICP4D Token Authentication
+  class ICP4DTokenManager < JWTTokenManager
+    def initialize(icp4d_url: nil, username: nil, password: nil, access_token: nil)
```

Comment:
Sorry - one more comment! Since this is already an ICP4D class, I think we should use the generic name `URL`. This is thinking forward to when the authentication methods become more generic in September.

Reply:
updated.